src/components/sbb-toggle/sbb-toggle.tsx

```diff
@@ -0,0 +1,200 @@
+  }
+
+  @Listen('did-select', { passive: true })
+  public onToggleOptionSelect(event: CustomEvent<Set<string>>): void {
+    this.value = event.detail;
+  }
+
+  private _setCheckedPillPosition(): void {
+    const checked = this._checked;
+
+    if (!checked) {
+      return;
+    }
+    this._hasCheckedOption = true;
+    this._element.style.setProperty('--sbb-toggle-option-left', `${checked?.offsetLeft - 2}px`);
+    this._element.style.setProperty(
+      '--sbb-toggle-option-right',
+      `${this._toggleElement.clientWidth - (checked?.offsetLeft + checked?.clientWidth) - 2}px`
+    );
+  }
+
+  public connectedCallback(): void {
+    this._toggleResizeObserver.observe(this._element.firstElementChild);
+  }
+
+  public disconnectedCallback(): void {
+    this._toggleResizeObserver.disconnect();
+  }
+
+  private _updateToggle(): void {
+    const options = this._options;
+    const value = this.value ?? options.find((toggleOption) => toggleOption.checked)?.value;
+
+    for (const toggleOption of options) {
+      toggleOption.checked = toggleOption.value === value;
+      toggleOption.disabled = toggleOption.disabled ? toggleOption.disabled : this.disabled;
+      toggleOption.tabIndex = this._getOptionTabIndex(toggleOption);
+    }
+
+    this._setFocusableOption();
+  }
+
+  private _setFocusableOption(): void {
+    if (!this._checked) {
+      this._options[0].tabIndex = 0;
+    }
+  }
+
+  private _getOptionTabIndex(option: HTMLSbbToggleOptionElement): number {
+    return option.checked && !option.disabled && !this.disabled ? 0 : -1;
+  }
+
+  @Listen('keydown')
+  public handleKeyDown(evt: KeyboardEvent): void {
+    if (!this._options) {
+      return;
+    }
+
+    const enabledToggleOptions = this._options.filter((t) => !t.disabled);
+    const checked = enabledToggleOptions.findIndex((radio) => radio.checked);
+    const cur = checked !== -1 ? checked : 0;
+    const size = enabledToggleOptions.length;
+    const prev = cur === 0 ? size - 1 : cur - 1;
+    const next = cur === size - 1 ? 0 : cur + 1;
+
+    // don't trap nested handling
+    if (
+      (evt.target as HTMLElement) !== this._element &&
+      (evt.target as HTMLElement).parentElement !== this._element
+    ) {
+      return;
+    }
+
+    const currentWritingMode = getDocumentWritingMode();
+    const prevKey = currentWritingMode === 'rtl' ? 'ArrowRight' : 'ArrowLeft';
+    const nextKey = currentWritingMode === 'rtl' ? 'ArrowLeft' : 'ArrowRight';
+
+    if (evt.key === prevKey || evt.key === 'ArrowUp') {
+      enabledToggleOptions[prev].select();
+      enabledToggleOptions[prev].focus();
+      evt.preventDefault();
+    } else if (evt.key === nextKey || evt.key === 'ArrowDown') {
+      enabledToggleOptions[next].select();
+      enabledToggleOptions[next].focus();
+      evt.preventDefault();
+    }
+  }
+
+  public render(): JSX.Element {
+    return (
+      <Host class={{ 'sbb-toggle--checked': this._hasCheckedOption }}>
```

Comment:
I'm not sure whether we should set classes on host. @kyubisation ? On the other hand, it can be nice as we use :host for real properties and can toggle variables on the same level.

Reply:
I'm a bit late on this; I'd say to generally avoid setting classes on host.
I'm also not sure how this applies here, as a state without a selection does not exist (according to spec).

Reply:
Agreed with avoiding classes on the host, and true, it's probably no longer necessary in this case (previously we handled the unselected status anyway, but now we always select the first option if none is specified). 👍